Enable failing diffs on regression

laithsakka opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #136551
* #136383

1. example of failing diff
https://github.com/pytorch/pytorch/pull/136740

2. test this by running 
python check_results.py test_check_result/expected_test.csv   test_check_result/result_test.csv 

results
```
WIN: benchmark ('a', ' instruction count') failed, actual result 90 is 18.18% lower than expected 110 ±1.00% please update the expected results.
REGRESSION: benchmark ('b', ' memory') failed, actual result 200 is 100.00% higher than expected 100 ±10.00% if this is an expected regression, please update the expected results.
MISSING REGRESSION TEST: benchmark ('d', ' missing-test') does not have a regression test enabled for it
```
MISSING REGRESSION TEST does not fail but its logged. 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

laithsakka committed:
enable failing diffs on regression

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"


regression introduced by setting low expected 
```
**REGRESSION** benchmark add_loop_eager_dynamic failed, actual instruction count 5486451976 is higher than expected 1 with noise margin 0.01 if this is an expected regression, please update the expected instruction count in the benchmark.
```

win introduced by setting high expected 
```
**WIN** benchmark add_loop_eager failed, actual instruction count 2758450573 is lower than expected 10000000000 with noise margin 0.01 please update the expected instruction count in the benchmark.
```
I will follow up with diffs that enable the regressions at diff time. 
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"


regression introduced by setting low expected 
```
**REGRESSION** benchmark add_loop_eager_dynamic failed, actual instruction count 5486451976 is higher than expected 1 with noise margin 0.01 if this is an expected regression, please update the expected instruction count in the benchmark.
```

win introduced by setting high expected 
```
**WIN** benchmark add_loop_eager failed, actual instruction count 2758450573 is lower than expected 10000000000 with noise margin 0.01 please update the expected instruction count in the benchmark.
```
I will follow up with diffs that enable the regressions at diff time. 
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"


regression introduced by setting low expected 
```
**REGRESSION** benchmark add_loop_eager_dynamic failed, actual instruction count 5486451976 is higher than expected 1 with noise margin 0.01 if this is an expected regression, please update the expected instruction count in the benchmark.
```

win introduced by setting high expected 
```
**WIN** benchmark add_loop_eager failed, actual instruction count 2758450573 is lower than expected 10000000000 with noise margin 0.01 please update the expected instruction count in the benchmark.
```
I will follow up with diffs that enable the regressions at diff time. 
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"


regression introduced by setting low expected 
```
**REGRESSION** benchmark add_loop_eager_dynamic failed, actual instruction count 5486451976 is higher than expected 1 with noise margin 0.01 if this is an expected regression, please update the expected instruction count in the benchmark.
```

win introduced by setting high expected 
```
**WIN** benchmark add_loop_eager failed, actual instruction count 2758450573 is lower than expected 10000000000 with noise margin 0.01 please update the expected instruction count in the benchmark.
```
I will follow up with diffs that enable the regressions at diff time. 
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"


regression introduced by setting low expected 
```
**REGRESSION** benchmark add_loop_eager_dynamic failed, actual instruction count 5486451976 is higher than expected 1 with noise margin 0.01 if this is an expected regression, please update the expected instruction count in the benchmark.
```

win introduced by setting high expected 
```
**WIN** benchmark add_loop_eager failed, actual instruction count 2758450573 is lower than expected 10000000000 with noise margin 0.01 please update the expected instruction count in the benchmark.
```
I will follow up with diffs that enable the regressions at diff time. 
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"


regression introduced by setting low expected 
```
**REGRESSION** benchmark add_loop_eager_dynamic failed, actual instruction count 5486451976 is higher than expected 1 with noise margin 0.01 if this is an expected regression, please update the expected instruction count in the benchmark.
```

win introduced by setting high expected 
```
**WIN** benchmark add_loop_eager failed, actual instruction count 2758450573 is lower than expected 10000000000 with noise margin 0.01 please update the expected instruction count in the benchmark.
```
I will follow up with diffs that enable the regressions at diff time. 
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"


regression introduced by setting low expected 
```
**REGRESSION** benchmark add_loop_eager_dynamic failed, actual instruction count 5486451976 is higher than expected 1 with noise margin 0.01 if this is an expected regression, please update the expected instruction count in the benchmark.
```

win introduced by setting high expected 
```
**WIN** benchmark add_loop_eager failed, actual instruction count 2758450573 is lower than expected 10000000000 with noise margin 0.01 please update the expected instruction count in the benchmark.
```
I will follow up with diffs that enable the regressions at diff time. 
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

ezyang commented:
You're putting the expected count number in a Python file, which means when you write the auto-updater it will be harder to programatically update. Can we plan for programattic updates now, or have you decided you don't want them?

oulgen commented:
Earlier today we discussed how you should log each PR you block so that we can know/track when a PR is blocked. Should we add that functionality before landing this?

ezyang commented:
Doesn't this PR have those logs?

laithsakka commented:
> Earlier today we discussed how you should log each PR you block so that we can know/track when a PR is blocked. Should we add that functionality before landing this?

yeh this PR have the logs cc @ezyang  @oulgen

laithsakka commented:
> You're putting the expected count number in a Python file, which means when you write the auto-updater it will be harder to programatically update. Can we plan for programattic updates now, or have you decided you don't want them?

mhmm i see i can change the way we do it to have them in a separate file

ezyang commented:
To be super explicit, I don't want to land this /without/ the autoupdater. It's a package deal.

laithsakka committed:
Update on "Enable failing diffs on regression"


regression introduced by setting low expected 
```
**REGRESSION** benchmark add_loop_eager_dynamic failed, actual instruction count 5486451976 is higher than expected 1 with noise margin 0.01 if this is an expected regression, please update the expected instruction count in the benchmark.
```

win introduced by setting high expected 
```
**WIN** benchmark add_loop_eager failed, actual instruction count 2758450573 is lower than expected 10000000000 with noise margin 0.01 please update the expected instruction count in the benchmark.
```
I will follow up with diffs that enable the regressions at diff time. 
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"


regression introduced by setting low expected 
```
**REGRESSION** benchmark add_loop_eager_dynamic failed, actual instruction count 5486451976 is higher than expected 1 with noise margin 0.01 if this is an expected regression, please update the expected instruction count in the benchmark.
```

win introduced by setting high expected 
```
**WIN** benchmark add_loop_eager failed, actual instruction count 2758450573 is lower than expected 10000000000 with noise margin 0.01 please update the expected instruction count in the benchmark.
```
I will follow up with diffs that enable the regressions at diff time. 
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"


test this by running 
python check_results.py test_check_result/expected_test.csv   test_check_result/result_test.csv 

results
```
WIN: benchmark ('a', ' instruction count') failed, actual result 90 is 18.18% lower than expected 110 ±1.00% please update the expected results.
REGRESSION: benchmark ('b', ' memory') failed, actual result 200 is 100.00% higher than expected 100 ±10.00% if this is an expected regression, please update the expected results.
MISSING REGRESSION TEST: benchmark ('d', ' missing-test') does not have a regression test enabled for it
```
MISSING REGRESSION TEST does not fail but its logged. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"


test this by running 
python check_results.py test_check_result/expected_test.csv   test_check_result/result_test.csv 

results
```
WIN: benchmark ('a', ' instruction count') failed, actual result 90 is 18.18% lower than expected 110 ±1.00% please update the expected results.
REGRESSION: benchmark ('b', ' memory') failed, actual result 200 is 100.00% higher than expected 100 ±10.00% if this is an expected regression, please update the expected results.
MISSING REGRESSION TEST: benchmark ('d', ' missing-test') does not have a regression test enabled for it
```
MISSING REGRESSION TEST does not fail but its logged. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"


test this by running 
python check_results.py test_check_result/expected_test.csv   test_check_result/result_test.csv 

results
```
WIN: benchmark ('a', ' instruction count') failed, actual result 90 is 18.18% lower than expected 110 ±1.00% please update the expected results.
REGRESSION: benchmark ('b', ' memory') failed, actual result 200 is 100.00% higher than expected 100 ±10.00% if this is an expected regression, please update the expected results.
MISSING REGRESSION TEST: benchmark ('d', ' missing-test') does not have a regression test enabled for it
```
MISSING REGRESSION TEST does not fail but its logged. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"


test this by running 
python check_results.py test_check_result/expected_test.csv   test_check_result/result_test.csv 

results
```
WIN: benchmark ('a', ' instruction count') failed, actual result 90 is 18.18% lower than expected 110 ±1.00% please update the expected results.
REGRESSION: benchmark ('b', ' memory') failed, actual result 200 is 100.00% higher than expected 100 ±10.00% if this is an expected regression, please update the expected results.
MISSING REGRESSION TEST: benchmark ('d', ' missing-test') does not have a regression test enabled for it
```
MISSING REGRESSION TEST does not fail but its logged. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"


test this by running 
python check_results.py test_check_result/expected_test.csv   test_check_result/result_test.csv 

results
```
WIN: benchmark ('a', ' instruction count') failed, actual result 90 is 18.18% lower than expected 110 ±1.00% please update the expected results.
REGRESSION: benchmark ('b', ' memory') failed, actual result 200 is 100.00% higher than expected 100 ±10.00% if this is an expected regression, please update the expected results.
MISSING REGRESSION TEST: benchmark ('d', ' missing-test') does not have a regression test enabled for it
```
MISSING REGRESSION TEST does not fail but its logged. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"


test this by running 
python check_results.py test_check_result/expected_test.csv   test_check_result/result_test.csv 

results
```
WIN: benchmark ('a', ' instruction count') failed, actual result 90 is 18.18% lower than expected 110 ±1.00% please update the expected results.
REGRESSION: benchmark ('b', ' memory') failed, actual result 200 is 100.00% higher than expected 100 ±10.00% if this is an expected regression, please update the expected results.
MISSING REGRESSION TEST: benchmark ('d', ' missing-test') does not have a regression test enabled for it
```
MISSING REGRESSION TEST does not fail but its logged. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka commented:
adress comments

ezyang commented on benchmarks/dynamo/pr_time_benchmarks/test_check_result/result_test.csv:
```diff
@@ -0,0 +1,4 @@
+a, instruction count, 90, 0.01
+b, memory, 200, 0.1
+c, something, 107, 0.1
+d, missing-test, 10
```
CSV with variable number of columns? 🤔

laithsakka replied:
whats your question? ah there should be only three i will remove 0.01 from the first row. 
this is am using to test my script.

ezyang commented on benchmarks/dynamo/pr_time_benchmarks/check_results.py:
```diff
@@ -0,0 +1,141 @@
+
+        if result < low:
+            fail = True
+            ratio = (float)(entry.expected_value - result) * 100 / entry.expected_value
```
the heck is this, just do `float(...)` like a normal person lol

laithsakka replied:
lol pardon my c++ backward your majesty i will fix it

ezyang commented on benchmarks/dynamo/pr_time_benchmarks/check_results.py:
```diff
@@ -0,0 +1,141 @@
+            print(
+                f"WIN: benchmark {key} failed, actual result {result} is {ratio:.2f}% lower than "
+                f"expected {entry.expected_value} ±{entry.noise_margin*100:.2f}% "
+                f"please update the expected results."
```
OK, so are you going to write the updater script too?

laithsakka replied:
not in this diff

laithsakka committed:
Update on "Enable failing diffs on regression"


1. example of failing diff
https://github.com/pytorch/pytorch/pull/136740

2. test this by running 
python check_results.py test_check_result/expected_test.csv   test_check_result/result_test.csv 

results
```
WIN: benchmark ('a', ' instruction count') failed, actual result 90 is 18.18% lower than expected 110 ±1.00% please update the expected results.
REGRESSION: benchmark ('b', ' memory') failed, actual result 200 is 100.00% higher than expected 100 ±10.00% if this is an expected regression, please update the expected results.
MISSING REGRESSION TEST: benchmark ('d', ' missing-test') does not have a regression test enabled for it
```
MISSING REGRESSION TEST does not fail but its logged. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka commented:
adress comments

laithsakka committed:
Update on "Enable failing diffs on regression"


1. example of failing diff
https://github.com/pytorch/pytorch/pull/136740

2. test this by running 
python check_results.py test_check_result/expected_test.csv   test_check_result/result_test.csv 

results
```
WIN: benchmark ('a', ' instruction count') failed, actual result 90 is 18.18% lower than expected 110 ±1.00% please update the expected results.
REGRESSION: benchmark ('b', ' memory') failed, actual result 200 is 100.00% higher than expected 100 ±10.00% if this is an expected regression, please update the expected results.
MISSING REGRESSION TEST: benchmark ('d', ' missing-test') does not have a regression test enabled for it
```
MISSING REGRESSION TEST does not fail but its logged. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on regression"


1. example of failing diff
https://github.com/pytorch/pytorch/pull/136740

2. test this by running 
python check_results.py test_check_result/expected_test.csv   test_check_result/result_test.csv 

results
```
WIN: benchmark ('a', ' instruction count') failed, actual result 90 is 18.18% lower than expected 110 ±1.00% please update the expected results.
REGRESSION: benchmark ('b', ' memory') failed, actual result 200 is 100.00% higher than expected 100 ±10.00% if this is an expected regression, please update the expected results.
MISSING REGRESSION TEST: benchmark ('d', ' missing-test') does not have a regression test enabled for it
```
MISSING REGRESSION TEST does not fail but its logged. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge
